fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""Utility class for plotting metrics with error ranges."""
+
+from logging import error
+from typing import Any, Dict, List, Union
+import pandas as pd
+
+from matplotlib.axes._axes import Axes
+
+from ._metric_frame import MetricFrame
+
+_MATPLOTLIB_IMPORT_ERROR_MESSAGE = "Please make sure to install matplotlib to use " \
+                                   "the plotting for MetricFrame."
+
+_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR = "Ambiguous Error Metric. Please only provide one of: error_bars and conf_intervals."
+
+_METRICS_NOT_LIST_OR_STR_ERROR = "Metric should be a string of a single metric or a list of the metrics in provided MetricFrame, but {0} was provided"
+_NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR = "upper_error, lower_error, and symmetric_error must be positive and upper_bound must be greater than lower_bound"
+_INVALID_ERROR_MAPPING = "Invalid error mapping, each metric must have all the key-value pairs that make up an Error Format"
+
+def plot_metric_frame(plot_type: str, metric_frame: MetricFrame, 
+                    metrics: Union[List[str], str]=None, error_bars: Union[List[str], str]=None, conf_intervals: Union[List[str], str]=None,
```

Comment:
metric_frame should be the only positional argument; and plot_type should be keyword, defaulting to a raster plot rather than bar plot.

Reply:
What is a raster plot in this context? Is it a scatter plot?